### ISSUE TITLE
CPAN::Meta::Prereqs prereq merging doesn't respect x_

### ISSUE THREAD

**Issue by dagolden** [Perl-Toolchain-Gang/CPAN-Meta#66]
The CMP constructor will accept "x_" fields, but 'with_merged_prereqs'
does not: it only takes "official" fields and it really take those, plus
any "x_" fields.

**Comment by karenetheridge** [comment]
> Yeah, methods doing list_phases and a relationships_for would be really helpful here.

In #110 I implemented these as private methods, but you're right that they would be useful more generally, so I'm going to modify those to make them public.

**Comment by karenetheridge** [comment]
new commit added to make these interfaces public. I'm not entirely sure that I picked the best names though -- is `my @phases = $prereqs->phases;` and `my @runtime_types = $prereqs->types_in('runtime');` clear enough?

**Comment by karenetheridge** [comment]
@haarg I'll remove the usage of List::Util::uniq.  Is this a suitable replacement?

```
sub __uniq {
  goto &List::Util::uniq if \&List::Util::uniq;
  keys %{ +{ map { $_ => undef } @_ } }
}
```

**Comment by xdg** [comment]
Probably fixed with #110
